src/handlers/get-base-rate-changes.ts

```diff
@@ -0,0 +1,74 @@
+import { Context } from "../types/context";
+import { Rates } from "../types/plugin-input";
+import { isPushEvent } from "../types/typeguards";
+
+/**
+ * Parses the diff of changes to the org config file to find the old and new base rates.
+ *
+ * This will capture changes to either the plugin's config or the global basePriceMultiplier.
+ */
+export async function getBaseRateChanges(context: Context): Promise<Rates> {
+  if (!isPushEvent(context)) {
+    context.logger.debug("Not a push event");
+    return {
+      previousBaseRate: null,
+      newBaseRate: null,
+    };
+  }
+  const {
+    logger,
+    payload: { repository, head_commit: headCommit },
+  } = context;
+  const commitSha = headCommit?.id;
+  let commitData;
+
+  if (!commitSha) {
+    throw new Error("No commit sha found");
+  }
+  const owner = repository.owner?.login
+
+  if (!owner) {
+    throw logger.error("No owner found in the repository");
+  }
+
+  try {
+    commitData = await context.octokit.repos.getCommit({
+      owner,
+      repo: repository.name,
+      ref: commitSha,
+      mediaType: {
+        format: "diff",
+      },
+    });
+  } catch (err: unknown) {
+    logger.debug("Commit sha error.", { err });
+  }
+
+  if (!commitData) {
+    throw new Error("No commit data found");
+  }
+
+  const data = commitData.data as unknown as string;
+  const changes = data.split("\n");
+
+  const newValue = /\+\s*basePriceMultiplier:\s*(\S+)/;
```

Comment:
This seems a bit dangerous and error prone if we change the name of the variable someday. Do you need to check the diff, can't we just always update even if the price is the same? What are the benefits from it?

Reply:
I see what you are saying. Other than not wasting 100s of our rate limit unnecessarily I can't think of anything immediately.

Without reading the diff we don't know if it's been changed so would that mean we run a global label update any time an admin/billing updates the config? I will come back to you on this

Reply:
Doesn't this run on an Action? I am not sure what is more costly between checking all the diffs or just updating the values. But if this is an Action we should be fine.